src/lib/FlightTasks/tasks/Utility/ObstacleAvoidance.cpp

```diff
@@ -101,7 +102,21 @@ void ObstacleAvoidance::injectAvoidanceSetpoints(Vector3f &pos_sp, Vector3f &vel
 	if (avoidance_data_timeout || !avoidance_point_valid) {
 		PX4_WARN("Obstacle Avoidance system failed, loitering");
 		_publishVehicleCmdDoLoiter();
+
+		if (!PX4_ISFINITE(_failsafe_position(0)) && !PX4_ISFINITE(_failsafe_position(1))
+		    && !PX4_ISFINITE(_failsafe_position(2))) {
```

Comment:
Even though it should never make a difference I would put `||` just in case there will ever be a corner case where only one is NAN.

Reply:
Was changed